README.md

```diff
@@ -35,17 +45,75 @@ To bypass the need for running locally AWS you can pass a global variable to the
+application we have added this to the `entrypoint.sh` file along with the `SCRIPT_NAME`.
+The Rails secret is also created here.
+
+There is a workaround to removing the PID lock of the Rails process in the event of
+the application crashing and not releasing the process.
+
+We have to pass the `API_SERVICE_URL` so that it is available in the `entrypoint.sh`
+or the application will throw an error and exit before starting
+
+### Prometheus monitoring
+
+[Prometheus](https://prometheus.io) is set up to provide metrics on the
+`/metrics` endpoint. The following metrics are recorded:
+
+- `data_api_status` (counter)
+  Response from data API
+- `data_api_connection_failure` (counter)
+  Could not connect to back-end data API'
+- `data_api_service_exception` (counter)
+  The response from the back-end data API was not processed'
+- `internal_application_error` (counter)
+  Unexpected events and internal error count
+- `memory_used_mb` (gauge)
+  Process memory usage in megabytes
+
+Internally, we use ActiveSupport Notifications to emit events which are
+monitored and collected by the Prometheus store. Relevant pieces of the
+app include:
+
+- `app/lib/prometheus/metrics.rb`
+  A central register where metrics can be stored and looked-up subsequently
+  by use of a symbol key.
+- `config/initializers/prometheus.rb`
+  Defines the Prometheus counters that the app knows about, and registers them
+  in the metrics store (see above)
+- `config/initializers/load_notification_subscribers.rb`
+  Some boiler-plate code to ensure that all of the notification subscribers
+  are loaded when the app starts
+- `app/subscribers`
+  Folder where the subscribers to the known ActiveSupport notifications are
+  defined. This is where the transform from `ActiveSupport::Notification` to
+  Prometheus counter or gauge is performed.
+
+In addition to the metrics we define, there is a collection of standard
+metrics provided automatically by the
+[Ruby Prometheus client](https://github.com/prometheus/client_ruby)
+
+To test Prometheus when developing locally, there needs to be a Prometheus
+server running. Tip for Linux users: do not install the Prometheus apt
+package. This starts a locally running daemon with a pre-defined
+configuration, which is useful when monitoring the machine on which the
+server is running. A better approach for testing Prometheus is to
+[download](https://prometheus.io/download/) the server package and
+run it locally, with a suitable configuration. A basic config for
+monitoring a Rails application is provided in `test/prometheus/dev.yml`.
+
+Using this approach, and assuming you install your local copy of
+Prometheus into `~/apps`, a starting command line would be something like:
+
+```sh
+~/apps/prometheus/prometheus-2.32.1.linux-amd64/prometheus \
+  --config.file=test/prometheus/dev.yml \
+  --storage.tsdb.path=./tmp/metrics2
+```
+
+Something roughly equivalent should be possible on Windows and Mac as well.
```

Comment:
I'm not sure this is necessary to star up a prometheus server in order to test the metrics I've certainly never done it for images I've produced, I simple curl has been sufficient. But now it's here then great.